index.html

```diff
@@ -1489,90 +1489,53 @@ <h2>Cryptographic Suites</h2>
-        </dd>
-        <dt><dfn>proof algorithm</dfn></dt>
-        <dd>
-An algorithm that takes an input message and produces an output value where the
-receiver of the message can mathematically verify that the message has not
-been modified in transit and came from someone possessing a particular secret.
-        </dd>
-      </dl>
 
       <p>
-At a minimum, a <a>proof type</a> is expected have the following attributes:
+The requirements for all data integrity cryptographic suite specifications are
+as follows:
       </p>
 
-      <dl style="margin-left: 1em;">
-        <dt>id</dt>
-        <dd>
-A URL that identifies the <a>cryptographic suite</a>.
-For example: <code>https://w3id.org/security#Ed25519Signature2020</code>.
-        </dd>
-        <dt>type</dt>
-        <dd>
-The value <code>ProofSuite</code>.
-        </dd>
-        <dt>canonicalizationAlgorithm</dt>
-        <dd>
-A URL that identifies the <a>canonicalization algorithm</a> to use on the
-<var>document</var>. For example:
-<code>https://w3id.org/security#URDNA2015</code>.
-        </dd>
-        <dt>digestAlgorithm</dt>
-        <dd>
-A URL that identifies the <a>message digest algorithm</a> to use on the
-<var>canonicalized document</var>. For example:
-<code>https://www.ietf.org/assignments/jwa-parameters#SHA256</code>
-        </dd>
-        <dt>proofAlgorithm</dt>
-        <dd>
-A URL that identifies the <a>proof algorithm</a> to use on the
-<var>data to be signed</var>. For example:
-<code>https://w3id.org/security#ed25519</code>
-        </dd>
-      </dl>
-
-      <p>
-A complete example of a <a>proof type</a> is shown in the next example:
-      </p>
+      <ul>
+        <li>
+The specification MUST be published as a human-readable document at a URL.
+        </li>
+        <li>
+The specification MUST identify a cryptographic suite <a>type</a> and any
+parameters that can be used with the suite.
+        </li>
+        <li>
+The specification MUST detail the algorithms used to perform transformations,
+if any, to the data to be protected.
+        </li>
+        <li>
+The specification MUST detail the algorithms used to perform cryptographic
+hashing to the data to be protected.
+        </li>
+        <li>
+The specification MUST detail the algorithms used to perform cryptographic
+protection of the data.
+        </li>
+        <li>
+The specification MUST detail the algorithms used to perform cryptographic
+verification of the data.
+        </li>
+        <li>
+The specification MUST contain a Security Considerations section detailing
+privacy considerations specific to the cryptographic suite.
+        </li>
+        <li>
+The specification MUST contain a Privacy Considerations section detailing
+privacy considerations specific to the cryptographic suite.
+        </li>
+        <li>
+The specification MUST provide an interoperability test suite to determine if
+a given implementation is conformant with the specification.
```

Comment:
I don't know how the specification provides this -- unless this refers to inline test vectors, perhaps. This language might need to be tweaked a little, but approving for now.

Reply:
Yes, you're right, the spec can't provide this. What I was trying to get to was some sort of higher bar for registration -- where you have to demonstrate some level of interoperability to get into the registry.

Perhaps what we should do here is strike the requirement and make it a part of the registration process OR say that the specification MUST point to an interoperability test suite report (which you can do via ReSpec).

Reply:
we ok with a SHOULD here?

Reply:
I'd personally be ok w/ a SHOULD, though it lowers the bar to getting an entry into the registry AND it puts the registry maintainers in the critical path wrt. making "judgement calls", which have gotten us in trouble in the past. Given that this text is already in an issue marker, noting that it's controversial, I'm going to stick with the MUST for now and see if we get serious push-back. The text in the issue marker now suggests:

> The specification MUST provide a link to an interoperability test report to document which implementations are conformant with the cryptographic suite specification.

Given the nature of cryptosuites, not providing such a test report (with a single entry in it) feels like a low enough bar already.